ONSdigital/rm-case-service
# Remove bi/disable enrolment - stop creating new B case

# Motivation and Context
Currently when we disable the final respondent for a business/survey we create a new B case. This creates a new enrolment code and adds a the new `B` case to the action service for future communications. A case should correspond to one response we want to receive so we should not be creating extra `B` cases on top of the existing ones for each business/collection exercise combination.

Instead we want the following behaviour in the case table when disabling the final respondent:
- No new B case is created
- Associated B case gets a new active enrolment code
- Associated B case is in `ACTIONABLE` state
- Associated B case is present in action service database (and old BI case is not)

# What has changed
Removed all logic associated with the `DISABLE_RESPONDENT_ENROLMENT` case event category. All this now does it update the status of the `BI` case to `INACTIONABLE`.
Added new category `NO_ACTIVE_ENROLMENTS` which replaces the enrolment code for the associated case and sets it's status to `ACTIONABLE` (which also removes the case from action service).

# How to test?
Run in docker with other services and associated PR's, check in database that when disabling the final respondent for a business/survey that case table is in correct state

On `mvn clean install`, should pass all unit/integration tests
Should pass acceptance tests when run alongside associated PR's

# Links
[Trello](https://trello.com/c/RlGDFpX8/18-disable-enrolment-stop-creating-b-cases-case-events-l)
## Related PR's
[casesvc-api](https://github.com/ONSdigital/rm-casesvc-api/pull/25)
[party-service](https://github.com/ONSdigital/ras-party/pull/149)

!!! This shouldn't be deployed into production until the appropriate data migration has been done to ensure that one `B` case exists for every casegroup !!!
